src/Core/src/Response.php

```diff
@@ -85,24 +86,74 @@ public function resolve(?float $timeout = null): bool
                 }
             }
 
-            $statusCode = $this->httpResponse->getStatusCode();
+            return $this->handleStatus();
         } catch (TransportExceptionInterface $e) {
             throw $this->resolveResult = new NetworkException('Could not contact remote server.', 0, $e);
         }
+    }
 
-        if (500 <= $statusCode) {
-            throw $this->resolveResult = new ServerException($this->httpResponse);
-        }
+    /**
+     * Make sure all provided requests are executed.
+     *
+     * @param self[]     $responses
+     * @param float|null $timeout   Duration in seconds before aborting. When null wait
+     *                              until the end of execution. Using 0 means non-blocking
+     *
+     * @return iterable<self>
+     *
+     * @throws NetworkException
+     * @throws HttpException
+     */
+    final public static function multiplex(iterable $responses, float $timeout = null): iterable
+    {
+        /** @var self[] $responseMap */
+        $responseMap = [];
+        $httpResponses = [];
+        $httpClient = null;
+        foreach ($responses as $response) {
+            if (null !== $response->resolveResult) {
+                yield $response;
+
+                continue;
+            }
 
-        if (400 <= $statusCode) {
-            throw $this->resolveResult = new ClientException($this->httpResponse);
+            if (null === $httpClient) {
+                $httpClient = $response->httpClient;
+            }
+            $httpResponses[] = $response->httpResponse;
+            $responseMap[\spl_object_id($response->httpResponse)] = $response;
         }
 
-        if (300 <= $statusCode) {
-            throw $this->resolveResult = new RedirectionException($this->httpResponse);
+        // response is empty
+        if (null === $httpClient) {
+            return;
         }
 
-        return $this->resolveResult = true;
+        try {
+            foreach ($httpClient->stream($httpResponses, $timeout) as $httpResponse => $chunk) {
+                if ($chunk->isTimeout()) {
+                    continue;
```

Comment:
This means all http responses are inactive. We should return here, not `continue`.

Reply:
Sure abut that? Can't we have a bunch of responses in the buffer, were the first is inactive and other unfetched buffer?

nb: I tested with `0` and I didn"t get an infinite loop, if you afraid about that

Reply:
From the docs: 

> If all responses become inactive for the given duration, the method will yield a special chunk whose isTimeout() will return true:

Reply:
fixed,thank for the pointer